### DISCUSSION TITLE
Help setting up Selenium

### DISCUSSION THREAD

**Discussion by rajp1503** [robbrad/UKBinCollectionData#751] [Q&A] [answered]
Now that I have Three Rivers District Council available in this integration, it requires a path to Selenium server.. I have no idea how I can set this up in my home assistant. Is there an easy step by step guide on this? I haven't seen it or just bad at doing a search. any help would be greatly appreciated. I run HA on in a VirtualBox image on my Windows 11 PC.

**Comment by dp247** [comment]
You could probably:
- Install Docker Desktop on Windows 11
- Get [standalone-chrome](https://hub.docker.com/r/selenium/standalone-chrome)
- Point HA to use it

**Reply by rajp1503** [reply]
I tried that but it still does not work.. most likely, I may not have setup something in the Stand Alone Chrome but as there are no instructions to validate that, I can't go further with that.

**Reply by robbrad** [reply]
What operating system are you on?

Are you running docker?

If you run Selenium Chrome Standalone can you access the selenium ui in your browser?

Depending on your home assistant setup (you mentioned a VM) can it access devices on your network can it access the selenium container via http?

Please let us know what you tried already.

We do have a small documentation gap on setting up selenium (more is available in contributing.md) but it's not really in the scope of this project to set it up, as there is plenty of online info (eg. https://medium.com/@sharmila.may5/steps-to-run-selenium-tests-in-docker-7610281a5581) on how to setup selenium. (Selenium while used by the project, is used for many other purposes online)

**Reply by robbrad** [reply]
@rajp1503 https://github.com/robbrad/UKBinCollectionData/blob/master/README.md#selenium

**Reply by rajp1503** [reply]
> @rajp1503 https://github.com/robbrad/UKBinCollectionData/blob/master/README.md#selenium

Thanks Rob... I have gone thru those steps, installed the container and checked that is running ok. Within the UK Bin integration, I have added the UPRN, Post Code and "http://localhost:4444/" as the URL... it seems to accept that info but doesn't provide anything else. What should I expect to see. It shows as No devices or Entities.

**Reply by robbrad** [reply, marked as answer]
You can put debug logging on for home assistant and see what the custom component is saying.

My gut feeling though is home assistant can't find localhost of where docker is running because it's in a VM

Instead of local host try the lan IP of the docker server.

**Reply by rajp1503** [reply]
> You can put debug logging on for home assistant and see what the custom component is saying.
> 
> My gut feeling though is home assistant can't find localhost of where docker is running because it's in a VM
> 
> Instead of local host try the lan IP of the docker server.

Thank you Rob... it was not able to translate the localhost. Once I put in the IP address, it worked straight away... much appreciated your help on this..

**Reply by robbrad** [reply]
That's good news

I would say have a think about your architecture here.

Ask yourself where you want to run stuff from.

Home assistant can also run as a container etc
